### PR TITLE
Refactor BatchUpdateJob

### PR DESCRIPTION
**Pull request by awead**
Allows for overriding BatchUpdateJob to inject additional jobs or additional metadata on each of the files in the batch.

### CODE REVIEW

File: sufia-models/app/jobs/batch_update_job.rb [hunk 1]
```diff
@@ -6,44 +7,45 @@ def queue_name
...
     if denied.empty?
-      send_user_success_message(user, batch) unless saved.empty?
+      send_user_success_message unless saved.empty?
     else
-      send_user_failure_message(user, batch)
+      send_user_failure_message
     end
   end
 
-  def update_file(gf, user)
```
Comment (jcoyne) [inline sufia-models/app/jobs/batch_update_job.rb]: Why are you changing signatures? That's not backwards compatible.

Comment (mjgiarlo) [inline sufia-models/app/jobs/batch_update_job.rb]: If these aren't private methods, perhaps they oughta be.

**Comment by awead** [comment]
I made it private, for @mjgiarlo, and added an optional user to the method params for @jcoyne.

File: sufia-models/app/jobs/batch_update_job.rb [hunk 2]
```diff
@@ -6,67 +7,103 @@ def queue_name
...
     message = denied.count > 1 ? multiple_failure(batch.id, denied) : single_failure(batch.id, denied.first)
     User.batchuser.send_message(user, message, failure_subject, false)
   end
+
+  private
+
+    def update_file(gf, you = nil)
+      you ||= user
+      unless you.can? :edit, gf
+        ActiveFedora::Base.logger.error "User #{you.user_key} DENIED access to #{gf.id}!"
```
Comment (grosscol) [inline sufia-models/app/jobs/batch_update_job.rb]: Will `#{you.user_key}` throw an error with when `you` is `nil`?  If so, why the default value of `nil` for `you`?

Comment (awead) [inline sufia-models/app/jobs/batch_update_job.rb]: If you is nil, it get's set to user on line 62. Right?

Comment (grosscol) [inline sufia-models/app/jobs/batch_update_job.rb]: misinterpreted that when I read it.

**Pull request merged**